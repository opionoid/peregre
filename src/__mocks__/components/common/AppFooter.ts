import { IAppFooterProps } from 'src/components/common/AppFooter';

const MOCK_APP_FOOTER_PROPS: IAppFooterProps = {};

export default MOCK_APP_FOOTER_PROPS;
