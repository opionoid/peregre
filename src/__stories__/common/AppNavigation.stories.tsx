import { Story, Meta } from '@storybook/react'
import {
  AppNavigation,
  IAppNavigationProps,
} from 'src/components/common/AppNavigation'
import MOCK_APP_NAVIGATION_PROPS from 'src/__mocks__/components/common/AppNavigation'
import { BrowserRouter as Router } from 'react-router-dom'

export default {
  title: 'components/common/AppNavigation',
  component: AppNavigation,
} as Meta

const Template: Story<IAppNavigationProps> = (args) => (
  <Router>
    <AppNavigation {...args} />
  </Router>
)
export const Default = Template.bind({})
Default.args = {
  ...MOCK_APP_NAVIGATION_PROPS,
}
