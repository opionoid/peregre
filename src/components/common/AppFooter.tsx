
import React from 'react'
import { Link } from 'react-router-dom'
import { color, space } from 'src/assets/style'
import { ROUTE } from 'src/constants'
import styled from 'styled-components'

export interface IAppFooterProps { }

export const AppFooter: React.VFC<IAppFooterProps> = () => {
  return (
    <footer>
      <FooterItem>
        <p>©︎ 2021, ことれの</p>
        <LinkList>
          <ListItem>
            <Link to={ROUTE.terms}>利用規約</Link>
          </ListItem>
          <ListItem>
            <Link to={ROUTE.privacy}>プライバシーポリシー</Link>
          </ListItem>
          <ListItem>
            <Link to="/">お問い合わせ</Link>
          </ListItem>
        </LinkList>
      </FooterItem>
    </footer>
  )
}

const FooterItem = styled.div`
  width: 100%;
  padding: 1.125em 0;
  background-color: ${color.backgroundHighContrast};
  text-align: center;
  color: ${color.fontInHighContrast};
`
const LinkList = styled.ul`
  display: flex;
  justify-content: center;
  align-items: center;
  list-style: none;
  column-gap: ${space.m}
`
const ListItem = styled.li`
  & > *,
  & > *:visited {
    color: ${color.fontInHighContrast};
    text-decoration: none;
    border: 0;
  }
`
