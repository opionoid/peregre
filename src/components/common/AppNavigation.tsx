import React from 'react'
import { Link } from 'react-router-dom'
import { useToggle } from 'react-use'
import { color } from 'src/assets/style'
import { ROUTE } from 'src/constants'
import styled, { css } from 'styled-components'

export interface IAppNavigationProps { }

export const AppNavigation: React.VFC<IAppNavigationProps> = () => {
  const [expanded, toggleHamburger] = useToggle(false)
  const handleClick = React.useCallback(() => toggleHamburger(), [toggleHamburger])
  const close = React.useCallback(() => toggleHamburger(false), [toggleHamburger])

  /** Chrome では Link に text-decoration: none を直接指定しない限り消えない（Safariは消える） */
  return (
    <nav>
      <AppNavigationWrapper className={expanded ? '-expanded' : ''}>
        <Top>
          <Link className="link" to={ROUTE.top} onClick={close} style={{ textDecoration: 'none' }}>
            Peregre
          </Link>
          <HamburgerMenu
            aria-expanded={expanded}
            onClick={handleClick}
          >
            {expanded ? <span aria-label='ナビゲーションを閉じる'>χ</span> : <span aria-label='ナビゲーションを開く'>ξ</span>}
          </HamburgerMenu>
        </Top>
        <li>
          <Links>
            <Item aria-expanded={!expanded}>
              <Link className="link" to={ROUTE.rules} onClick={close} style={{ textDecoration: 'none' }}>
                RULES
              </Link>
            </Item>
            <Item aria-expanded={!expanded}>
              <Link className="link" to={ROUTE.character} onClick={close} style={{ textDecoration: 'none' }}>
                CHARACTER
              </Link>
            </Item>
            <Item aria-expanded={!expanded}>
              <Link className="link" to={ROUTE.stories} onClick={close} style={{ textDecoration: 'none' }}>
                STORIES
              </Link>
            </Item>
            <Item aria-expanded={!expanded}>
              <Link className="link" to={ROUTE.news} onClick={close} style={{ textDecoration: 'none' }}>
                NEWS
              </Link>
            </Item>
          </Links>
        </li>
        <Config aria-expanded={!expanded}>
          <Link className="link" to={ROUTE.config} onClick={close} style={{ textDecoration: 'none' }}>
            Config
          </Link>
        </Config>
      </AppNavigationWrapper>
    </nav>
  )
}

const AppNavigationWrapper = styled.ul`
  width: 100%;
  text-decoration: none;
  display: flex;
  text-align: center;
  justify-content: space-between;
  padding-inline-start: 0;
  height: fit-content;
  border-bottom: 1px solid ${color.accent};
  list-style: none;
  position: fixed;
  backdrop-filter: blur(4px) grayscale(0.03);
  z-index: 100;
  margin-top: 0;

  @media screen and (max-width: 46em) {
    display: block;
    width: 100%;
    border: 0;
    margin: 0;
    height: 3.8125em;

    &.-expanded {
      min-height: 99vh;
      background-color: rgba(254, 248, 231, 0.86);
    }
  }
`
const Links = styled.ul`
  display: flex;
  list-style: none;
  padding: 0;

  @media screen and (max-width: 46em) {
    display: block;
    width: 100%;
  }
`

const ItemCss = css`
  &:hover {
    > a.link {
      color: ${color.fontInHighContrast};
      border-bottom: 0;
      background-color: ${color.backgroundHighContrast};
    }
  }

  > a.link {
    font-size: 1.125em;
    display: block;
    padding: 0.85em 1.7em;
    color: ${color.font};
    border-bottom: 0;
    font-family: 'Helvetica Neue' sans-serif;

    @media screen and (max-width: 46em) {
      width: 100%;
      padding: 0.85em 0;
    }
  }
`
const Top = styled.li`
  ${ItemCss};
  font-style: italic;
  position: relative;

  @media screen and (max-width: 46em) {
    margin-bottom: 25vh;
    padding-right: 4em;
    width: calc(100% - 4em);
  }
`
const HamburgerMenu = styled.button`
  display: none;

  @media screen and (max-width: 46em) {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    width: 4rem;
    height: 61px;
    border: 0;
    border-bottom: 1px solid ${color.accent};
    color: inherit;
    font-size: 1.125em;
    padding: 1.115em 1.5em;
    background-color: inherit;
    cursor: pointer;

    &:hover {
      background-color: ${color.accent};
    }
  }
`
const Item = styled.li`
  ${ItemCss};

  @media screen and (max-width: 46em) {
    &[aria-expanded='true'] {
      display: none;
    }
  }
`
const Config = styled.li`
  ${ItemCss};
  font-style: italic;

  @media screen and (max-width: 46em) {
    margin-top: 2.25em;

    &[aria-expanded='true'] {
      display: none;
    }
  }
`
