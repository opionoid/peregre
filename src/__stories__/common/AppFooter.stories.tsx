import { Story, Meta } from '@storybook/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { AppFooter, IAppFooterProps } from 'src/components/common/AppFooter'
import MOCK_APP_FOOTER_PROPS from 'src/__mocks__/components/common/AppFooter'

export default {
  title: 'components/common/AppFooter',
  component: AppFooter,
} as Meta

const Template: Story<IAppFooterProps> = (args) => (
  <Router>
    <AppFooter {...args} />
  </Router>
)
export const Default = Template.bind({})
Default.args = {
  ...MOCK_APP_FOOTER_PROPS
}