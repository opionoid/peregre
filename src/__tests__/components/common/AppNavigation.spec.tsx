import { BrowserRouter as Router } from 'react-router-dom'
import { AppNavigation } from 'src/components/common/AppNavigation'
import MOCK_APP_NAVIGATION_PROPS from 'src/__mocks__/components/common/AppNavigation'
import { render } from '@testing-library/react'

describe('common', () => {
  const snapshot = render(
    <Router>
      <AppNavigation {...MOCK_APP_NAVIGATION_PROPS} />
    </Router>
  )

  it('default test', () => {
    expect(snapshot.baseElement).toMatchSnapshot()
  })
})
