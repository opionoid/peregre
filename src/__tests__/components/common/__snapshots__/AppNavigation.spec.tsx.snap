// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common default test 1`] = `
<body>
  <div>
    <nav>
      <ul
        class="sc-bdvvaa fEaskq"
      >
        <li
          class="sc-dkPtyc hWcuzK"
        >
          <a
            class="link"
            href="/"
            style="text-decoration: none;"
          >
            Peregre
          </a>
          <button
            aria-expanded="false"
            class="sc-hKwCoD ffYizP"
          >
            Î¾
          </button>
        </li>
        <li>
          <ul
            class="sc-gsDJrp kwhPwZ"
          >
            <li
              aria-expanded="true"
              class="sc-eCImvq iDUuly"
            >
              <a
                class="link"
                href="/rules"
                style="text-decoration: none;"
              >
                RULES
              </a>
            </li>
            <li
              aria-expanded="true"
              class="sc-eCImvq iDUuly"
            >
              <a
                class="link"
                href="/character"
                style="text-decoration: none;"
              >
                CHARACTER
              </a>
            </li>
            <li
              aria-expanded="true"
              class="sc-eCImvq iDUuly"
            >
              <a
                class="link"
                href="/stories"
                style="text-decoration: none;"
              >
                STORIES
              </a>
            </li>
            <li
              aria-expanded="true"
              class="sc-eCImvq iDUuly"
            >
              <a
                class="link"
                href="/news"
                style="text-decoration: none;"
              >
                NEWS
              </a>
            </li>
          </ul>
        </li>
        <li
          aria-expanded="true"
          class="sc-jRQAMF kstTgy"
        >
          <a
            class="link"
            href="/config"
            style="text-decoration: none;"
          >
            Config
          </a>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
